src/otx/core/metrics/dice.py

```diff
@@ -2,16 +2,33 @@
 # SPDX-License-Identifier: Apache-2.0
 #
 """Module for OTX Dice metric used for the OTX semantic segmentation task."""
+from torch import Tensor
+from torchmetrics import JaccardIndex
 from torchmetrics.classification.dice import Dice
 from torchmetrics.collections import MetricCollection
 
 from otx.core.types.label import LabelInfo
 
 
-def _dice_callable(label_info: LabelInfo) -> MetricCollection:
+def _segm_callable(label_info: LabelInfo) -> MetricCollection:
     return MetricCollection(
-        {"Dice": Dice(num_classes=label_info.num_classes + 1, ignore_index=label_info.num_classes)},
+        {
+            "Dice": OTXDice(num_classes=label_info.num_classes, ignore_index=-1, average="macro"),
+            "mIoU": JaccardIndex(task="multiclass", num_classes=label_info.num_classes, ignore_index=255),
```

Comment:
Why should they have different values for `ignore_index`?

Reply:
In addition, does we already has a dedicated `label_info` metadata entity, `SegLabelInfo`? By following your explanations,

> Ignore index was handled wrongly. Loss function also used 255 ignore index ...

I think that `ignore_index` in the metric should be configurable and `label_info` can be a source of that configuration. This time you can add `ignore_index` field to `SegLabelInfo` and set its default value as 255.

Reply:
It is different, because Dice metric from torchvision doesn't support 255 common ignore index and raises an error regarding this. Moreover, it doesn't support even -1, it just ignores it w/o errors. 
Implementation for this class is not so good. Copy and paste class constructor just to delete assertion seemed to me not a good idea, so I use this workaround

Reply:
Ok, I will try to add possibility to change ignore index, however, we should also change loss function and augmentations

Reply:
As you mentioned, it is configurable from the model (the loss function) and data pipeline (the augmentations). My hope is that it should be configurable at the metric side as same as the others, even if we normaly use a fixed value for `ignore_index=255` in fact.